src/otx/algo/instance_segmentation/heads/bbox_head.py

```diff
@@ -391,3 +374,306 @@ def export_by_feat(
+        additional parameters pos_inds_list and neg_inds_list to
+        `_get_targets_single` function.
+
+        Args:
+            sampling_results (list[obj:SamplingResult]): Assign results of
+                all images in a batch after sampling.
+            batch_img_metas (list[dict]): Meta information of each image, e.g., image size, scaling factor, etc.
+            concat (bool): Whether to concatenate the results of all
+                the images in a single batch.
+
+        Returns:
+            tuple[Tensor]: Ground truth for proposals in a single image.
+            Containing the following list of Tensors:
+
+            - labels (list[Tensor],Tensor): Gt_labels for all
+                proposals in a batch, each tensor in list has
+                shape (num_proposals,) when `concat=False`, otherwise
+                just a single tensor has shape (num_all_proposals,).
+            - label_weights (list[Tensor]): Labels_weights for
+                all proposals in a batch, each tensor in list has
+                shape (num_proposals,) when `concat=False`, otherwise
+                just a single tensor has shape (num_all_proposals,).
+            - bbox_targets (list[Tensor],Tensor): Regression target
+                for all proposals in a batch, each tensor in list
+                has shape (num_proposals, 4) when `concat=False`,
+                otherwise just a single tensor has shape
+                (num_all_proposals, 4), the last dimension 4 represents
+                [tl_x, tl_y, br_x, br_y].
+            - bbox_weights (list[tensor],Tensor): Regression weights for
+                all proposals in a batch, each tensor in list has shape
+                (num_proposals, 4) when `concat=False`, otherwise just a
+                single tensor has shape (num_all_proposals, 4).
+        """
+        pos_priors_list = [res.pos_priors for res in sampling_results]
+        neg_priors_list = [res.neg_priors for res in sampling_results]
+        pos_gt_bboxes_list = [res.pos_gt_bboxes for res in sampling_results]
+        pos_gt_labels_list = [res.pos_gt_labels for res in sampling_results]
+        labels, label_weights, bbox_targets, bbox_weights = multi_apply(
+            self._get_targets_single,
+            pos_priors_list,
+            neg_priors_list,
+            pos_gt_bboxes_list,
+            pos_gt_labels_list,
+        )
+
+        valid_label_mask = self.get_valid_label_mask(img_metas=batch_img_metas, all_labels=labels, use_bg=True)
+        valid_label_mask = [i.to(labels[0].device) for i in valid_label_mask]
+
+        if concat:
+            labels = torch.cat(labels, 0)
+            label_weights = torch.cat(label_weights, 0)
+            bbox_targets = torch.cat(bbox_targets, 0)
+            bbox_weights = torch.cat(bbox_weights, 0)
+            valid_label_mask = torch.cat(valid_label_mask, 0)
+        return labels, label_weights, bbox_targets, bbox_weights, valid_label_mask
+
+
+class BBoxHead:
```

Comment:
You left previously comment about "ConvFCBBoxHead" unappropriated naming and I changed it to simple one
![image](https://github.com/user-attachments/assets/a0725151-c7de-4395-b995-424df3df78b1)

Reply:
Ah, that comment was the suggestion about using factory names to ~builder instead of module's name as is.
Because some parts need module's name to check some conditions such as xai.

Reply:
Renamed back https://github.com/openvinotoolkit/training_extensions/pull/3865/commits/3a68981ed9d3cbc18f1d6c9b84cf6de1b9edff5b

Reply:
Renaming all factory to names with builder suffix  should be considered for all tasks. We may consider that